packages/did/src/Did.chain.ts

```diff
@@ -598,14 +569,20 @@ export async function getSetKeyExtrinsic(
   key: NewDidKey
 ): Promise<Extrinsic> {
   const api = await BlockchainApiConnection.getConnectionOrConnect()
-  const keyAsEnum = formatPublicKey(key)
+  function requireVerificationKey(): PublicKeyEnum<NewDidVerificationKey> {
```

Comment:
if you look at it, it checks that the key is a verification key (not encryption key). We can also not do that, in which case polkadot will simply complain that it can't encode if you give an inconsistent key type/key role combination to this function (e.g. an x25519 key as authentication key). Means less informative errors though. Also wasn't a fan of doing this check before the switch, because that means it would complain about the wrong key type before even checking which role it is for (although all the roles currently allowed in this function require verification keys)

Reply:
Yes, that’s what I am saying, all the roles require verification keys, so the check could be done earlier. By the way, the getSetKeyExtrinsic could request the relationship to be of type `VerificationKeyRelationship`, would that make you reconsider the earlier type check?